var/spack/repos/builtin/packages/ufs-weather-model/package.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class UfsWeatherModel(CMakePackage):
+    """The Unified Forecast System (UFS) Weather Model (WM) is a prognostic
+    model that can be used for short- and medium-range research and
+    operational forecasts, as exemplified by its use in the operational Global
+    Forecast System (GFS) of the National Oceanic and Atmospheric
+    Administration (NOAA)."""
+
+    homepage = "https://ufs-weather-model.readthedocs.io/en/latest/"
+    url      = "https://github.com/ufs-community/ufs-weather-model/archive/refs/tags/ufs-v1.1.0.tar.gz"
+    git      = "https://github.com/ufs-community/ufs-weather-model.git"
+
+    maintainers = ['t-brown']
+
+    version('2.0.0', tag='ufs-v2.0.0', submodules=True)
+    version('1.1.0', tag='ufs-v1.1.0', submodules=True)
+
+    variant('32bit', default=True, description='Enable 32-bit single precision arithmetic in dycore')
+    variant('avx2', default=False, description='Enable AVX2 instructions')
+    variant('build_type', default='Release', description='CMake build type',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
+    variant('ccpp', default=True, description='Enable the Common Community Physics Package (CCPP))')
+    variant('ccpp_suites', default='FV3_GFS_v15p2,FV3_RRFS_v1alpha',
+            description='CCPP suites to compile',
+            values=('FV3_GFS_v15p2',
+                    'FV3_RRFS_v1alpha',
+                    'FV3_GFS_v15p2,FV3_RRFS_v1alpha'),
+            multi=True)
+    variant('inline_post', default=False, description='Compile post processing inline')
+    variant('multi_gases', default=False, description='Enable multi gases in physics routines')
+    variant('openmp', default=True, description='Enable OpenMP')
+    variant('parallel_netcdf', default=True, description='Enable parallel I/O in netCDF')
+    variant('quad_precision', default=False,
+            description='Enable quad precision for certain grid metric terms in dycore')
+    variant('simdmultiarch', default=False, description='Enable multi-target SIMD instruction sets')
+
+    depends_on('bacio')
+    depends_on('esmf@:8.0.0')
+    depends_on('mpi')
+    depends_on('nemsio')
+    depends_on('netcdf-c')
+    depends_on('netcdf-fortran')
+    depends_on('sp')
+    depends_on('w3emc')
+    depends_on('w3nco')
+
+    def setup_build_environment(self, env):
+        spec = self.spec
+        env.set('CMAKE_C_COMPILER', spec['mpi'].mpicc)
+        env.set('CMAKE_CXX_COMPILER', spec['mpi'].mpicxx)
+        env.set('CMAKE_Fortran_COMPILER', spec['mpi'].mpifc)
```

Comment:
Do these need to be configured in two places?

Reply:
They do not. They have to be set as `env` vars, due to the brain-dead way they wrote their `CMakeLists.txt`. I'll remove them from the `args` and hope upstream changes in the next release.